src/utils/gui/globjects/GUIPolygon.cpp

```diff
@@ -150,11 +153,22 @@ TesselatedPolygon::drawTesselation(const PositionVector& shape) const {
             gluTessVertex(tobj, points + 3 * i, points + 3 * i);
         }
         gluTessEndContour(tobj);
-
+        size_t startIndex = shape.size() * 3;
+        for (size_t j = 0; j < myHoles.size(); j++) {
+            PositionVector hole = myHoles[j];
+            gluTessBeginContour(tobj);
+            for (int i = 0; i < (int)hole.size(); i++) {
+                points[startIndex + 3 * i] = hole[i].x();
+                points[startIndex + 3 * i + 1] = hole[i].y();
+                points[startIndex + 3 * i + 2] = 0.;
+                gluTessVertex(tobj, points + startIndex + 3 * i, points + startIndex + 3 * i);
```

Comment:
here as well. Please iterate over the vector and increment the startIndex in the loop